app/logic/certification.py

```diff
@@ -79,28 +88,49 @@ def updateCertRequirements(certId, newRequirements):
         A list of CertificationRequirement objects corresponding to the given `newRequirements` list.
     """
     # check for missing ids to remove
-    saveIds = [requirementData['id'] for requirementData in newRequirements]
-    CertificationRequirement.delete().where(CertificationRequirement.certification_id == certId, CertificationRequirement.id.not_in(saveIds)).execute()
+    # saveIds = [int(requirementData['id']) for requirementData in newRequirements]
+
 
     # update existing and add new requirements
-    requirements = []
-    for order, requirementData in enumerate(newRequirements):
-        try:
-            newRequirement = CertificationRequirement.get_by_id(requirementData['id'])
-        except DoesNotExist:
-            newRequirement = CertificationRequirement()
 
+    if newRequirements.get("save-new",None) is not None:
```

Comment:
From what I understand in the docstring, newRequirements should be a list of dictionaries, which you do not reflect on this line. It seems to me that you are trying to use it as a list only, not a list of dictionaries.